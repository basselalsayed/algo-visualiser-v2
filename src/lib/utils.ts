import { type ClassValue, clsx } from 'clsx';
import { twMerge } from 'tailwind-merge';

export function cn(...inputs: ClassValue[]) {
  return twMerge(clsx(inputs));
}

export function sleep(ms: number): Promise<void> {
  return new Promise((resolve) => setTimeout(resolve, ms));
}

type T0 =
  | 'string'
  | 'number'
  | 'bigint'
  | 'boolean'
  | 'symbol'
  | 'undefined'
  | 'object'
  | 'function';

export function assert(value: unknown, type: T0): asserts value is T0 {
  const actualType = typeof value;

  if (type === 'object') {
    if (value === null) {
      throw new TypeError(`Expected type '${type}', but received null`);
    }
  }

  if (actualType !== type) {
    throw new TypeError(
      `Expected type '${type}', but received '${actualType}'`
    );
  }
}

export function typeIs(value: unknown, type: T0): value is T0 {
  try {
    assert(value, type);
    return true;
  } catch {
    return false;
  }
}

export function convertToMilliseconds(
  start: DOMHighResTimeStamp,
  end: DOMHighResTimeStamp
) {
  const milliseconds = end - start;
  return Math.floor(milliseconds * 1000) / 1000; // 3 decimal places
}

export function convertToSeconds(
  start: DOMHighResTimeStamp,
  end: DOMHighResTimeStamp
) {
  const seconds = (end - start) / 1000;
  return Math.floor(seconds * 1000) / 1000;
}

export function elementIsFullyScrolled(element: HTMLElement): boolean {
  return (
    Math.floor(element.scrollHeight - element.scrollTop) ===
    Math.floor(element.clientHeight)
  );
}
