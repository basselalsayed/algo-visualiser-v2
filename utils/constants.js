import path from 'node:path';

export const DIRECTORIES = [
  'algorithms',
  'components',
  'contexts',
  'data',
  'hooks',
  'lib',
  'store',
];

export const ROOT_PATH = path.resolve(import.meta.dirname, '..');
