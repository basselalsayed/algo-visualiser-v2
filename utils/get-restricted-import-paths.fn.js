import fs from 'node:fs';
import path from 'node:path';

const __dirname = new URL('.', import.meta.url).pathname;

/**
 * Recursively walks through a directory and collects relative paths to all index.ts or index.tsx files.
 *
 * @param {string} dir - Absolute path to start walking from.
 * @param {string} baseRelativePath - Relative path from the module root (used for nested structure).
 * @returns {string[]} - List of relative paths to directories containing index files.
 */
function findIndexFileDirs(dir, baseRelativePath = '') {
  const indexDirs = [];
  const entries = fs.readdirSync(dir);

  for (const entry of entries) {
    const fullPath = path.join(dir, entry);
    const relPath = path.join(baseRelativePath, entry);
    const stat = fs.statSync(fullPath);

    if (stat.isDirectory()) {
      indexDirs.push(...findIndexFileDirs(fullPath, relPath));
    } else if (
      (entry === 'index.ts' || entry === 'index.tsx') &&
      baseRelativePath !== ''
    ) {
      indexDirs.push(baseRelativePath);
    }
  }

  return indexDirs;
}

/**
 * Generates a list of restricted import paths for all index.ts(x) files in a module.
 *
 * @param {string} dirAbsolutePath - Absolute path to the module directory (e.g. src/components)
 * @param {string} alias - Import alias for the module (e.g. @/components)
 * @returns {Array<{ name: string, message: string }>}
 */
function getRestrictedIndexImports(dirAbsolutePath, alias) {
  const indexDirs = findIndexFileDirs(dirAbsolutePath);

  const staticUpwardPaths = [
    '../../../index',
    '../../..',
    '../../index',
    '../..',
    '../index',
    '..',
  ].map((name) => ({
    message: `Avoid importing '${name}'. Use aliased or direct file paths.`,
    name,
  }));

  const indexPathViolations = indexDirs.flatMap((relPath) => [
    {
      message: `Avoid importing index file: ${alias}/${relPath}/index.ts. Import specific files instead.`,
      name: `${alias}/${relPath}`,
    },
    {
      message: `Avoid importing index file: ../${relPath}/index.ts. Import specific files instead.`,
      name: `../${relPath}`,
    },
  ]);

  return [...staticUpwardPaths, ...indexPathViolations];
}

export { __dirname, getRestrictedIndexImports };
