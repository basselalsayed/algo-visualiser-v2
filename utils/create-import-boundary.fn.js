import path from 'node:path';

import { DIRECTORIES, ROOT_PATH } from './constants.js';
import { getRestrictedIndexImports } from './get-restricted-import-paths.fn.js';

/**
 * @param {string} dir
 */
const getAlias = (dir) => `@/${dir}`;

/**
 * @param {string} dir
 */
export function createImportBoundary(dir) {
  const alias = getAlias(dir);
  const glob = `src/${dir}/**/*.{ts,tsx}`;

  const forbid = DIRECTORIES.flatMap((_dir) =>
    _dir === dir ? undefined : [`${getAlias(_dir)}/**/*`, `${getAlias(_dir)}/*`]
  ).filter(Boolean);

  const absoluteDir = path.resolve(ROOT_PATH, 'src', dir);

  const restrictedIndexPaths = getRestrictedIndexImports(absoluteDir, alias);

  return {
    files: [glob],
    rules: {
      'import/no-internal-modules': [
        'error',
        {
          forbid,
        },
      ],
      'no-restricted-imports': [
        'error',
        {
          paths: [
            ...restrictedIndexPaths,
            {
              message: `Avoid importing '${alias}' from within ${dir}/. Use direct file paths.`,
              name: alias,
            },
          ],
          patterns: [
            {
              group: ['**/index', '**/index.ts', '**/index.tsx'],
              message: `Avoid importing index files directly. Import specific files instead.`,
            },
            {
              group: ['../../**/*'],
              message: `Avoid relative imports outside of this module. Use alias paths instead.`,
            },
          ],
        },
      ],
    },
  };
}
