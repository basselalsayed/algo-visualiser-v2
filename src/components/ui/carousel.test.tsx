import { renderFragmentSnapshot } from '@/test/utils';

import {
  Carousel,
  CarouselContent,
  CarouselDotButtons,
  CarouselItem,
  CarouselNext,
  CarouselPrevious,
} from './carousel';

describe('Carousel UI snapshot', () => {
  it('looks the same', () => {
    expect(
      renderFragmentSnapshot(
        <Carousel>
          <CarouselContent>
            <CarouselItem>Slide 1</CarouselItem>
            <CarouselItem>Slide 2</CarouselItem>
            <CarouselItem>Slide 3</CarouselItem>
          </CarouselContent>
          <CarouselPrevious />
          <CarouselNext />
          <CarouselDotButtons />
        </Carousel>
      )
    ).toMatchSnapshot();
  });
});
