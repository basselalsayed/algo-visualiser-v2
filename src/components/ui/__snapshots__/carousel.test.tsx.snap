// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Carousel UI snapshot > looks the same 1`] = `
<DocumentFragment>
  <div
    aria-roledescription="carousel"
    class="relative"
    role="region"
  >
    <div
      class="space-y-4 overflow-hidden"
    >
      <div
        class="flex -ml-4"
        style="transform: translate3d(NaNpx,0px,0px);"
      >
        <div
          aria-roledescription="slide"
          class="min-w-0 shrink-0 grow-0 basis-full pl-4"
          role="group"
        >
          Slide 1
        </div>
        <div
          aria-roledescription="slide"
          class="min-w-0 shrink-0 grow-0 basis-full pl-4"
          role="group"
        >
          Slide 2
        </div>
        <div
          aria-roledescription="slide"
          class="min-w-0 shrink-0 grow-0 basis-full pl-4"
          role="group"
        >
          Slide 3
        </div>
      </div>
      <div
        class="flex items-center justify-between"
      >
        <div
          class="space-x-2"
        >
          <button
            class="inline-flex items-center justify-center gap-2 text-base font-medium whitespace-nowrap transition-colors focus-visible:ring-1 focus-visible:ring-ring focus-visible:outline-hidden disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 animate_grad bg_grad_accent--outline--text shadow-sm [--base-color:hsl(var(--background))] h-8 w-8 rounded-full"
            disabled=""
          >
            <svg
              class="h-4 w-4"
              fill="none"
              height="15"
              viewBox="0 0 15 15"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M6.85355 3.14645C7.04882 3.34171 7.04882 3.65829 6.85355 3.85355L3.70711 7H12.5C12.7761 7 13 7.22386 13 7.5C13 7.77614 12.7761 8 12.5 8H3.70711L6.85355 11.1464C7.04882 11.3417 7.04882 11.6583 6.85355 11.8536C6.65829 12.0488 6.34171 12.0488 6.14645 11.8536L2.14645 7.85355C1.95118 7.65829 1.95118 7.34171 2.14645 7.14645L6.14645 3.14645C6.34171 2.95118 6.65829 2.95118 6.85355 3.14645Z"
                fill="currentColor"
                fill-rule="evenodd"
                stroke="url(#gradient-mock-id)"
              />
              <defs>
                <lineargradient
                  gradientUnits="userSpaceOnUse"
                  id="gradient-mock-id"
                  style="--grad-1: hsl(var(--grad-accent-1)); --grad-2: hsl(var(--grad-accent-2));"
                  x1="0%"
                  x2="100%"
                  y1="0%"
                  y2="100%"
                >
                  <stop
                    offset="0%"
                    style="stop-color: var(--grad-1);"
                  />
                  <stop
                    offset="100%"
                    style="stop-color: var(--grad-2);"
                  />
                </lineargradient>
              </defs>
            </svg>
            <span
              class="sr-only"
            >
              sr.prevSlide
            </span>
          </button>
          <button
            class="inline-flex items-center justify-center gap-2 text-base font-medium whitespace-nowrap transition-colors focus-visible:ring-1 focus-visible:ring-ring focus-visible:outline-hidden disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 animate_grad bg_grad_accent--outline--text shadow-sm [--base-color:hsl(var(--background))] h-8 w-8 rounded-full"
            disabled=""
          >
            <svg
              class="h-4 w-4"
              fill="none"
              height="15"
              viewBox="0 0 15 15"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M8.14645 3.14645C8.34171 2.95118 8.65829 2.95118 8.85355 3.14645L12.8536 7.14645C13.0488 7.34171 13.0488 7.65829 12.8536 7.85355L8.85355 11.8536C8.65829 12.0488 8.34171 12.0488 8.14645 11.8536C7.95118 11.6583 7.95118 11.3417 8.14645 11.1464L11.2929 8H2.5C2.22386 8 2 7.77614 2 7.5C2 7.22386 2.22386 7 2.5 7H11.2929L8.14645 3.85355C7.95118 3.65829 7.95118 3.34171 8.14645 3.14645Z"
                fill="currentColor"
                fill-rule="evenodd"
                stroke="url(#gradient-mock-id)"
              />
              <defs>
                <lineargradient
                  gradientUnits="userSpaceOnUse"
                  id="gradient-mock-id"
                  style="--grad-1: hsl(var(--grad-accent-1)); --grad-2: hsl(var(--grad-accent-2));"
                  x1="0%"
                  x2="100%"
                  y1="0%"
                  y2="100%"
                >
                  <stop
                    offset="0%"
                    style="stop-color: var(--grad-1);"
                  />
                  <stop
                    offset="100%"
                    style="stop-color: var(--grad-2);"
                  />
                </lineargradient>
              </defs>
            </svg>
            <span
              class="sr-only"
            >
              sr.nextSlide
            </span>
          </button>
        </div>
        <div
          class="flex justify-center gap-x-2"
        />
      </div>
    </div>
    <button
      class="inline-flex items-center justify-center gap-2 text-base font-medium whitespace-nowrap transition-colors focus-visible:ring-1 focus-visible:ring-ring focus-visible:outline-hidden disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 animate_grad bg_grad_accent--outline--text shadow-sm [--base-color:hsl(var(--background))] h-8 w-8 rounded-full"
      disabled=""
    >
      <svg
        class="h-4 w-4"
        fill="none"
        height="15"
        viewBox="0 0 15 15"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M6.85355 3.14645C7.04882 3.34171 7.04882 3.65829 6.85355 3.85355L3.70711 7H12.5C12.7761 7 13 7.22386 13 7.5C13 7.77614 12.7761 8 12.5 8H3.70711L6.85355 11.1464C7.04882 11.3417 7.04882 11.6583 6.85355 11.8536C6.65829 12.0488 6.34171 12.0488 6.14645 11.8536L2.14645 7.85355C1.95118 7.65829 1.95118 7.34171 2.14645 7.14645L6.14645 3.14645C6.34171 2.95118 6.65829 2.95118 6.85355 3.14645Z"
          fill="currentColor"
          fill-rule="evenodd"
          stroke="url(#gradient-mock-id)"
        />
        <defs>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="gradient-mock-id"
            style="--grad-1: hsl(var(--grad-accent-1)); --grad-2: hsl(var(--grad-accent-2));"
            x1="0%"
            x2="100%"
            y1="0%"
            y2="100%"
          >
            <stop
              offset="0%"
              style="stop-color: var(--grad-1);"
            />
            <stop
              offset="100%"
              style="stop-color: var(--grad-2);"
            />
          </lineargradient>
        </defs>
      </svg>
      <span
        class="sr-only"
      >
        sr.prevSlide
      </span>
    </button>
    <button
      class="inline-flex items-center justify-center gap-2 text-base font-medium whitespace-nowrap transition-colors focus-visible:ring-1 focus-visible:ring-ring focus-visible:outline-hidden disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 animate_grad bg_grad_accent--outline--text shadow-sm [--base-color:hsl(var(--background))] h-8 w-8 rounded-full"
      disabled=""
    >
      <svg
        class="h-4 w-4"
        fill="none"
        height="15"
        viewBox="0 0 15 15"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M8.14645 3.14645C8.34171 2.95118 8.65829 2.95118 8.85355 3.14645L12.8536 7.14645C13.0488 7.34171 13.0488 7.65829 12.8536 7.85355L8.85355 11.8536C8.65829 12.0488 8.34171 12.0488 8.14645 11.8536C7.95118 11.6583 7.95118 11.3417 8.14645 11.1464L11.2929 8H2.5C2.22386 8 2 7.77614 2 7.5C2 7.22386 2.22386 7 2.5 7H11.2929L8.14645 3.85355C7.95118 3.65829 7.95118 3.34171 8.14645 3.14645Z"
          fill="currentColor"
          fill-rule="evenodd"
          stroke="url(#gradient-mock-id)"
        />
        <defs>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="gradient-mock-id"
            style="--grad-1: hsl(var(--grad-accent-1)); --grad-2: hsl(var(--grad-accent-2));"
            x1="0%"
            x2="100%"
            y1="0%"
            y2="100%"
          >
            <stop
              offset="0%"
              style="stop-color: var(--grad-1);"
            />
            <stop
              offset="100%"
              style="stop-color: var(--grad-2);"
            />
          </lineargradient>
        </defs>
      </svg>
      <span
        class="sr-only"
      >
        sr.nextSlide
      </span>
    </button>
    <div
      class="flex justify-center gap-x-2"
    />
  </div>
</DocumentFragment>
`;

exports[`Carousel UI snapshot > renders correctly 1`] = `
<DocumentFragment>
  <div
    aria-roledescription="carousel"
    class="relative"
    role="region"
  >
    <div
      class="space-y-4 overflow-hidden"
    >
      <div
        class="flex -ml-4"
        style="transform: translate3d(NaNpx,0px,0px);"
      >
        <div
          aria-roledescription="slide"
          class="min-w-0 shrink-0 grow-0 basis-full pl-4"
          role="group"
        >
          Slide 1
        </div>
        <div
          aria-roledescription="slide"
          class="min-w-0 shrink-0 grow-0 basis-full pl-4"
          role="group"
        >
          Slide 2
        </div>
        <div
          aria-roledescription="slide"
          class="min-w-0 shrink-0 grow-0 basis-full pl-4"
          role="group"
        >
          Slide 3
        </div>
      </div>
      <div
        class="flex items-center justify-between"
      >
        <div
          class="space-x-2"
        >
          <button
            class="inline-flex items-center justify-center gap-2 text-base font-medium whitespace-nowrap transition-colors focus-visible:ring-1 focus-visible:ring-ring focus-visible:outline-hidden disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 animate_grad bg_grad_accent--outline--text shadow-sm [--base-color:hsl(var(--background))] h-8 w-8 rounded-full"
            disabled=""
          >
            <svg
              class="h-4 w-4"
              fill="none"
              height="15"
              viewBox="0 0 15 15"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M6.85355 3.14645C7.04882 3.34171 7.04882 3.65829 6.85355 3.85355L3.70711 7H12.5C12.7761 7 13 7.22386 13 7.5C13 7.77614 12.7761 8 12.5 8H3.70711L6.85355 11.1464C7.04882 11.3417 7.04882 11.6583 6.85355 11.8536C6.65829 12.0488 6.34171 12.0488 6.14645 11.8536L2.14645 7.85355C1.95118 7.65829 1.95118 7.34171 2.14645 7.14645L6.14645 3.14645C6.34171 2.95118 6.65829 2.95118 6.85355 3.14645Z"
                fill="currentColor"
                fill-rule="evenodd"
                stroke="url(#gradient-mock-id)"
              />
              <defs>
                <lineargradient
                  gradientUnits="userSpaceOnUse"
                  id="gradient-mock-id"
                  style="--grad-1: hsl(var(--grad-accent-1)); --grad-2: hsl(var(--grad-accent-2));"
                  x1="0%"
                  x2="100%"
                  y1="0%"
                  y2="100%"
                >
                  <stop
                    offset="0%"
                    style="stop-color: var(--grad-1);"
                  />
                  <stop
                    offset="100%"
                    style="stop-color: var(--grad-2);"
                  />
                </lineargradient>
              </defs>
            </svg>
            <span
              class="sr-only"
            >
              sr.prevSlide
            </span>
          </button>
          <button
            class="inline-flex items-center justify-center gap-2 text-base font-medium whitespace-nowrap transition-colors focus-visible:ring-1 focus-visible:ring-ring focus-visible:outline-hidden disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 animate_grad bg_grad_accent--outline--text shadow-sm [--base-color:hsl(var(--background))] h-8 w-8 rounded-full"
            disabled=""
          >
            <svg
              class="h-4 w-4"
              fill="none"
              height="15"
              viewBox="0 0 15 15"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M8.14645 3.14645C8.34171 2.95118 8.65829 2.95118 8.85355 3.14645L12.8536 7.14645C13.0488 7.34171 13.0488 7.65829 12.8536 7.85355L8.85355 11.8536C8.65829 12.0488 8.34171 12.0488 8.14645 11.8536C7.95118 11.6583 7.95118 11.3417 8.14645 11.1464L11.2929 8H2.5C2.22386 8 2 7.77614 2 7.5C2 7.22386 2.22386 7 2.5 7H11.2929L8.14645 3.85355C7.95118 3.65829 7.95118 3.34171 8.14645 3.14645Z"
                fill="currentColor"
                fill-rule="evenodd"
                stroke="url(#gradient-mock-id)"
              />
              <defs>
                <lineargradient
                  gradientUnits="userSpaceOnUse"
                  id="gradient-mock-id"
                  style="--grad-1: hsl(var(--grad-accent-1)); --grad-2: hsl(var(--grad-accent-2));"
                  x1="0%"
                  x2="100%"
                  y1="0%"
                  y2="100%"
                >
                  <stop
                    offset="0%"
                    style="stop-color: var(--grad-1);"
                  />
                  <stop
                    offset="100%"
                    style="stop-color: var(--grad-2);"
                  />
                </lineargradient>
              </defs>
            </svg>
            <span
              class="sr-only"
            >
              sr.nextSlide
            </span>
          </button>
        </div>
        <div
          class="flex justify-center gap-x-2"
        />
      </div>
    </div>
    <button
      class="inline-flex items-center justify-center gap-2 text-base font-medium whitespace-nowrap transition-colors focus-visible:ring-1 focus-visible:ring-ring focus-visible:outline-hidden disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 animate_grad bg_grad_accent--outline--text shadow-sm [--base-color:hsl(var(--background))] h-8 w-8 rounded-full"
      disabled=""
    >
      <svg
        class="h-4 w-4"
        fill="none"
        height="15"
        viewBox="0 0 15 15"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M6.85355 3.14645C7.04882 3.34171 7.04882 3.65829 6.85355 3.85355L3.70711 7H12.5C12.7761 7 13 7.22386 13 7.5C13 7.77614 12.7761 8 12.5 8H3.70711L6.85355 11.1464C7.04882 11.3417 7.04882 11.6583 6.85355 11.8536C6.65829 12.0488 6.34171 12.0488 6.14645 11.8536L2.14645 7.85355C1.95118 7.65829 1.95118 7.34171 2.14645 7.14645L6.14645 3.14645C6.34171 2.95118 6.65829 2.95118 6.85355 3.14645Z"
          fill="currentColor"
          fill-rule="evenodd"
          stroke="url(#gradient-mock-id)"
        />
        <defs>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="gradient-mock-id"
            style="--grad-1: hsl(var(--grad-accent-1)); --grad-2: hsl(var(--grad-accent-2));"
            x1="0%"
            x2="100%"
            y1="0%"
            y2="100%"
          >
            <stop
              offset="0%"
              style="stop-color: var(--grad-1);"
            />
            <stop
              offset="100%"
              style="stop-color: var(--grad-2);"
            />
          </lineargradient>
        </defs>
      </svg>
      <span
        class="sr-only"
      >
        sr.prevSlide
      </span>
    </button>
    <button
      class="inline-flex items-center justify-center gap-2 text-base font-medium whitespace-nowrap transition-colors focus-visible:ring-1 focus-visible:ring-ring focus-visible:outline-hidden disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 animate_grad bg_grad_accent--outline--text shadow-sm [--base-color:hsl(var(--background))] h-8 w-8 rounded-full"
      disabled=""
    >
      <svg
        class="h-4 w-4"
        fill="none"
        height="15"
        viewBox="0 0 15 15"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M8.14645 3.14645C8.34171 2.95118 8.65829 2.95118 8.85355 3.14645L12.8536 7.14645C13.0488 7.34171 13.0488 7.65829 12.8536 7.85355L8.85355 11.8536C8.65829 12.0488 8.34171 12.0488 8.14645 11.8536C7.95118 11.6583 7.95118 11.3417 8.14645 11.1464L11.2929 8H2.5C2.22386 8 2 7.77614 2 7.5C2 7.22386 2.22386 7 2.5 7H11.2929L8.14645 3.85355C7.95118 3.65829 7.95118 3.34171 8.14645 3.14645Z"
          fill="currentColor"
          fill-rule="evenodd"
          stroke="url(#gradient-mock-id)"
        />
        <defs>
          <lineargradient
            gradientUnits="userSpaceOnUse"
            id="gradient-mock-id"
            style="--grad-1: hsl(var(--grad-accent-1)); --grad-2: hsl(var(--grad-accent-2));"
            x1="0%"
            x2="100%"
            y1="0%"
            y2="100%"
          >
            <stop
              offset="0%"
              style="stop-color: var(--grad-1);"
            />
            <stop
              offset="100%"
              style="stop-color: var(--grad-2);"
            />
          </lineargradient>
        </defs>
      </svg>
      <span
        class="sr-only"
      >
        sr.nextSlide
      </span>
    </button>
    <div
      class="flex justify-center gap-x-2"
    />
  </div>
</DocumentFragment>
`;
